### ISSUE TITLE
[Bug Report]: VZM31-SN BLUE 2-1 - 2.15 - HA - Minimum Dimming Level not honored

### ISSUE THREAD

**Issue by AzGilrock** [InovelliUSA/Firmware#7]
### What product is the bug report for?

Blue Series 2-1 Switch (VZM31-SN)

### FW Version

2.15

### Hub

Home Assistant 

### Bug Report Details

I bought a refurbished blue 2-1 switch a few days ago.  Last night I was prompted to update firmware.  It went from 16908800 to 16908815 which I’m assuming is 2.00 to 2.15.  I’ve re-synced all parameters and checked them and I’ve restarted zigbee2mqtt.   I have no automations setup.   When I hold the down paddle the brightness setting is going down past my minimum setting of 60.  Usually ends up at 2 but I’ve seen it stop at 3.  The minimum level was working as expected in the 2.00 FW.

For the screenshot below I did a double tap up which set it to 178 and then I held the down paddle until it quit changing and then let go.   You can see in the state the brightness is a 2.  That left the light barely visible well below what it should be if it stopped at 60.

I’m a Principal Software Engineer so I was hoping to review the code when I got to this repo but it appears to only have the actual FW files.

 
![image](https://github.com/InovelliUSA/Firmware/assets/2024669/0e920435-16fb-4788-818b-2258c43c10c4)

### How To Reproduce

Start from any state above the minimum and hold the down paddle.   The brightness is going below the minimum level.   I had my wife hold the button while I  state and I saw the brightness hit several spots below 60 before she released the button.   It ended on 2.

**Comment by InovelliUSA** [comment]
Hi @AzGilrock, the min and max level work internally on the switch. The switch will always be able to go from 0-254, but internally there is a scaled limit of whatever min and max are set to.

**Comment by AzGilrock** [comment]
Right I believe I understand how it’s supposed to work.  If I hold the down paddle it shouldn’t be taking the bulb down to where it’s almost off min level is set to 23%.  I thought the whole idea of minimum seating was so a guest couldn’t dim a light too far low or if a bulb flickers at low values.  It should only go below min when ramping off.   And like I said it seemed to work that way in the earlier FW.  I can take a video another day if needed.  Also I ordered 3 more so I can compare and see if they all work the same.  


I have double tap down set to the same as the min so when it dims too low I can double tap down and it brightens back up.

**Comment by AzGilrock** [comment]
The bulbs are dumb LED.  ExoSmart from Home Depot and they said Dimmable on the box. I can try the factory reset again. I think last one was before the Fw update.  So if I set Min to 60 and the Double Tap Down to 60 will the double tap and a Dimming action both land on the same brightness?   Or is the Double Tap value scaled inside the Min Max.

**Comment by InovelliUSA** [comment]
If you set min to 60 then 60 will be the new 1. So if you double tap down to 60 you will be going to a higher light level. Kind of confusing but, there is a pretty good explanation in the link below, but it is for z-wave. The same applies for the zigbee switch but the scale is 0-254 instead of 0-99.

https://community.inovelli.com/t/firmware-v1-57-production-lzw31-sn-dimmer-red-series-gen-2/8572/60

**Comment by AzGilrock** [comment]
Ok thanks.  Well that would explain why my double tap is raising the brightness.  I figured it was scaling the value I see in the Home Assistant GUI like the sliders but I didn't know the adjustment applied to the internal settings also.   I was trying to make my double tap down the same brightness as my min so I guess I set it to 0 to do that.  What's throwing me is I initially set the Min by moving the slider in HA and then looked at the number and it was like 23% so I calculated the value to enter.  It sure seems like its letting me dim way lower that what I initially had come up with but I'll try a reset and retest things.

**Comment by AzGilrock** [comment]
Ok I factory reset and did a lot of testing.  I think everything is ok now that I understand the double tap setting uses the scaled range.  Also the min brightness I like turned out to be 90.  So I changed from a smart bulb to a dumb bulb in the middle of my setup since I decided with the switch I didn't need a smart bulb. So 60 was a good min for the smart bulb.  So what was happening was I use down paddle to dim and it looked way too low.  Then I would double tap down and it would brighten up to what I expected.  Both the min and double tap were set to 60.  So the double tap was scaling it up and that means when you change the max it also was adjusting the brightness of the double tap down.  Anyways sorry for the confusion I'll close the issue.
